src/App.ts

```diff
@@ -7,83 +7,193 @@ import * as bodyParser from "body-parser";
-    (req: express.Request, res: express.Response) => {
-      logger.info("Serving Payment Check Request (GET)...");
-      toExpressHandler(
-        PaymentController.checkPaymentToPagoPa(
-          config.PAGOPA,
-          verificaRPTPagoPaClient
-        )
-      )(req, res);
-    }
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_REQUESTS_GET,
+    getPaymentInfoHandler
+  );
+
+  const activatePaymentHandler = toExpressHandler(
+    PaymentController.activatePayment(config.PAGOPA, pagoPAClient)
   );
   app.post(
-    config.CONTROLLER.ROUTES.RESTFUL.PAYMENTS_ACTIVATION,
-    (req: express.Request, res: express.Response) => {
-      logger.info("Serving Payment Activation Request (POST)...");
-      toExpressHandler(
-        PaymentController.activatePaymentToPagoPa(
-          config.PAGOPA,
-          attivaRPTPagoPaClient
-        )
-      )(req, res);
-    }
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_ACTIVATIONS_POST,
+    jsonParser,
+    activatePaymentHandler
+  );
+
+  const getActivationStatusHandler = toExpressHandler(
+    PaymentController.getActivationStatus(redisClient)
   );
+  app.get(
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_ACTIVATIONS_GET,
+    getActivationStatusHandler
+  );
+
+  // Endpoint for OpenAPI handler
+  app.get("/specs/api/v1/swagger.json", GetOpenapi(publicApiV1Specs));
+}
+
+/**
+ * Define a redis client necessary to handle persistent data
+ * @param {Configuration} config - Server Configuration
+ * @return {(app: core.Express) => soap.Server} A method to execute for start server listening
+ */
+function getRedisClient(config: Configuration): redis.RedisClient {
+  if (!config.REDIS_DB.USE_CLUSTER) {
+    logger.debug("Creating a REDIS client...");
+    return redis.createClient(config.REDIS_DB.HOST);
+  }
+  logger.debug("Creating a REDIS client using cluster...");
+  return new RedisClustr({
+    redisOptions: {
+      auth_pass: config.REDIS_DB.PASSWORD,
+      tls: {
+        servername: config.REDIS_DB.HOST
+      }
+    },
+    servers: [
+      {
+        host: config.REDIS_DB.HOST,
+        port: config.REDIS_DB.PORT
+      }
+    ]
+  }) as redis.RedisClient;
+  return redis.createClient(config.REDIS_DB.PORT);
+}
+
+/**
+ * Define a Service Handler for FespCdService SOAP service
+ * It's an endpoint for PagoPA to confirm a payment activation result
+ * @param {redis.RedisClient} redisClient - Redis Client to store persistente paymentId into DB
+ * @param {number} redisTimeoutSecs - Timeout set for information stored into DB
+ * @return {IFespCdPortTypeSoap} An object containing the service handler
+ */
+function getFespCdServiceHandler(
+  redisClient: redis.RedisClient,
+  redisTimeoutSecs: number
+): IFespCdPortTypeSoap {
+  return {
+    cdInfoPagamento: (
+      iCdInfoPagamentoInput: IcdInfoPagamentoInput,
+      callback: (
+        err: any | null, // tslint:disable-line:no-any
+        iCdInfoPagamentoOutput: IcdInfoPagamentoOutput,
+        raw: string,
```

Comment:
since you don't use `raw` and `soapHeader`, I believe you can avoid declaring them in the `callback` type

Reply:
It's defined\exposed with these args on IFespCdPortTypeSoap. How can I avoid it?

Reply:
ok it's fine to leave it